test/common/upstream/subset_lb_test.cc

```diff
@@ -1942,15 +1954,19 @@ TEST_F(SubsetLoadBalancerTest, EnabledScaleLocalityWeights) {
   // locality to be selected less because we've excluded more hosts in that locality than in the
   // first.
   // The localities are split 50/50, but because of the scaling we expect to see 66/33 instead.
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[0][1], lb_->chooseHost(&context));
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[1][3], lb_->chooseHost(&context));
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[0][1], lb_->chooseHost(&context));
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[0][1], lb_->chooseHost(&context));
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[1][3], lb_->chooseHost(&context));
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[0][1], lb_->chooseHost(&context));
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[0][1], lb_->chooseHost(&context));
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[1][3], lb_->chooseHost(&context));
-  EXPECT_EQ(host_set_.healthy_hosts_per_locality_->get()[0][1], lb_->chooseHost(&context));
+  const size_t iters = WRSQScheduler<int>::accuracy();
+  std::vector<size_t> counts({0, 0});
+  for (size_t i = 0; i < iters; ++i) {
+    EXPECT_CALL(random_, random()).WillRepeatedly(Return(i));
+    auto h = lb_->chooseHost(&context);
+    if (h == host_set_.healthy_hosts_per_locality_->get()[0][1]) {
+      counts[0]++;
+    } else if (h == host_set_.healthy_hosts_per_locality_->get()[1][3]) {
+      counts[1]++;
+    }
+  }
+  EXPECT_NEAR(2.0 * iters / 3, counts[0], 1);
+  EXPECT_NEAR(iters / 3.0, counts[1], 1);
```

Comment:
The determinism in this loop comes from random() returning the positive integers in order.  You could do an exact expectation.  When iters is a multiple of 3:
EXPECT_EQ(2 * iters / 3, counts[0]);
EXPECT_EQ(iters / 3, counts[1]);